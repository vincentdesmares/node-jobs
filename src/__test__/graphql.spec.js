const request = require('supertest')
const testDatabase = require('./test-database.js')
const migrateDatabase = testDatabase.migrateDatabase
const seedDatabase = testDatabase.seedDatabase
const getNewServer = testDatabase.getNewServer

const server = getNewServer()

/**
 * Starting the tests
 */
describe('Test the graphql queries', () => {
  beforeAll(async () => {
    await migrateDatabase()
    await seedDatabase()
    await server.start()
  })

  afterAll(async () => {
    await server.stop()
  })

  it('QUERY: projects', async () => {
    const response = await request(server.getGraphqlServer()).get(
      `/graphql?query=query getJobs {
              job {
                id
                name    
              }
        }
        &operationName=getJobs`
    )

    expect(response.body).toMatchSnapshot()
  })

  //   it("QUERY: project 1", async () => {
  //     const response = await request(server.getGraphqlServer()).get(
  //       singleProject(1)
  //     );

  //     expect(response.body).toMatchSnapshot();
  //   });

  //   it("QUERY: project 0 non existent", async () => {
  //     const response = await request(server.getGraphqlServer()).get(
  //       singleProject(0)
  //     );

  //     expect(response.body.data.project).toEqual(expect.arrayContaining([]));
  //   });

  //   it("QUERY: project events of project 1 when organizer", async () => {
  //     const response = await request(server.getGraphqlServer()).get(
  //       projectEvents(1)
  //     );

  //     expect(response.body).toMatchSnapshot();
  //   });

  //   it("QUERY: project events of project 1 when not organizer", async () => {
  //     const response = await request(server.getGraphqlServer())
  //       .get(projectEvents(1))
  //       .set("userid", 3);
  //     expect(response.body).toMatchSnapshot();
  //   });

  //   it("QUERY: project events of project 1", async () => {
  //     const response = await request(server.getGraphqlServer())
  //       .get(projectEvents(1))
  //       .set("userid", 3); // Non-organizers should not see the draft or cancelled events
  //     expect(response.body).toMatchSnapshot();
  //   });

  //   it("QUERY: project events of project 0 non existent", async () => {
  //     const response = await request(server.getGraphqlServer()).get(
  //       projectEvents(0)
  //     );

  //     expect(response.body.data.projectEvent).toEqual(expect.arrayContaining([]));
  //   });

  //   it("QUERY: get all users belonging to the same organization", async () => {
  //     const response = await request(server.getGraphqlServer()).get(users());

  //     expect(response.body).toMatchSnapshot();
  //   });

  //   it("QUERY: projects are not available to disabled users", async () => {
  //     const response = await request(server.getGraphqlServer())
  //       .get(projects())
  //       .set("userid", 6);
  //     expect(response.body).toMatchSnapshot();
  //   });
})
