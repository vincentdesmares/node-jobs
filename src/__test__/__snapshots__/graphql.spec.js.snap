// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the graphql queries QUERY: projects 1`] = `"1"`;
